CMakeLists.txt

```diff
@@ -1,22 +1,58 @@
-cmake_minimum_required(VERSION 2.8.9)
-project(concordpp)
+cmake_minimum_required(VERSION 3.1.3 FATAL_ERROR)
+
+# Force external build because source tree builds make a mess.
+if(${CMAKE_CURRENT_SOURCE_DIR} STREQUAL ${CMAKE_CURRENT_BINARY_DIR} AND NOT WIN32)
+	    message(FATAL_ERROR "You can not use CMake to build from the root of it's source tree! Remove the CMakeCache.txt file from this directory, then create a separate directory (either below this directory or elsewhere), and then re-run CMake from there.")
+endif(${CMAKE_CURRENT_SOURCE_DIR} STREQUAL ${CMAKE_CURRENT_BINARY_DIR} AND NOT WIN32)
+
+project(concordpp CXX)
+enable_language(C)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 #set(CMAKE_BUILD_TYPE Release)
+# Use boost multithreading
+set(Boost_USE_MULTITHREADED  ON)
 find_package(Boost 1.58 REQUIRED COMPONENTS system)
+
 find_package(OpenSSL REQUIRED openssl)
+find_package(CURL REQUIRED)
+# Required because boost/C++11 uses pthread
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
 
 # Disable tests for JSON
 set(BuildTests OFF CACHE INTERNAL "" FORCE)
```

Comment:
not review related but I this can be deleted if I'm correct, since nlohmann/json is only used as its raw header file instead of full repo w/ CMakeLists.txt.